### PR TITLE
fix(user): can not update is_active field via massive action

### PR DESCRIPTION
**Pull request by Rom1-B**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !35356
- Here is a brief description of what this PR does

Following https://github.com/glpi-project/glpi/pull/18441, the `is_active` field can still be modified by massive actions, a success message is displayed saying that the user has been modified, but the field was not modified in base. This PR is designed to display an error message while maintaining this security control.

### CODE REVIEW

**Comment by trasher** [comment]
Please a dedicated test

**Comment by Rom1-B** [comment]
> Please a dedicated test

There's already a test dedicated to Sensitive Fields, which I've updated, is that enough?

**Comment by trasher** [comment]
> > Please a dedicated test
> 
> There's already a test dedicated to Sensitive Fields, which I've updated, is that enough?

Seems good, yes :)

**Review by trasher — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
If I recall correctly, the field can only be updated for users that have less rights than you ?

In this case, maybe the error message should indicate the names of the users, so it is clear which user have been modified and which have not.

**Comment by Rom1-B** [comment]
It's already listed (_31688 is the user name I can't change_).

image

File: src/User.php [hunk 1]
```diff
@@ -1076,9 +1076,28 @@ public function prepareInputForUpdate($input)
                 count(array_intersect($protected_input_keys, array_keys($input))) > 0
                 && !$this->currentUserHaveMoreRightThan($input['id'])
             ) {
+                $ignored_fields = [];
                 foreach ($protected_input_keys as $input_key) {
+                    if (
+                        isset($input[$input_key])
+                        && $input_key != '_useremails' // Always in $input
```
Comment (cedric-anne) [inline src/User.php:1083]: All virtual fields should be excluded from this comparison.

```suggestion
                        && !str_starts_with($input_key, '_') // virtual field
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Pull request merged**